libmachine/provision/ubuntu.go

```diff
@@ -47,36 +47,55 @@ func (provisioner *UbuntuProvisioner) Service(name string, action pkgaction.Serv
 }
 
 func (provisioner *UbuntuProvisioner) Package(name string, action pkgaction.PackageAction) error {
-	var (
-		packageAction  string
-		updateMetadata = true
-	)
+	var packageAction string
+
+	updateMetadata := true
 
 	switch action {
-	case pkgaction.Install:
+	case pkgaction.Install, pkgaction.Upgrade:
 		packageAction = "install"
 	case pkgaction.Remove:
 		packageAction = "remove"
 		updateMetadata = false
-	case pkgaction.Upgrade:
-		packageAction = "upgrade"
 	}
 
-	// TODO: This should probably have a const
 	switch name {
 	case "docker":
-		name = "lxc-docker"
+		name = "docker-engine"
 	}
 
 	if updateMetadata {
-		// issue apt-get update for metadata
-		if _, err := provisioner.SSHCommand("sudo -E apt-get update"); err != nil {
+		if _, err := provisioner.SSHCommand("sudo apt-get update"); err != nil {
 			return err
 		}
 	}
 
+	// handle the new docker-engine package; we can probably remove this
+	// after we have a few versions
+	if action == pkgaction.Upgrade && name == "docker-engine" {
+		// run the force remove on the existing lxc-docker package
+		// and remove the existing apt source list
+		// also re-run the get.docker.com script to properly setup
+		// the system again
+
+		commands := []string{
+			"rm /etc/apt/sources.list.d/docker.list || true",
+			"apt-get remove -y lxc-docker || true",
+			"curl -sSL https://get.docker.com | sh",
```

Comment:
Longer-term, I'm wondering if it might not be worthwhile to replace that script with Go code in machine, but for now it Gets The Job Done.